webui/src/pages/repositories/wizard/wizard.jsx

```diff
@@ -1,100 +1,107 @@
-                    onNextStep = () => {},
-                    children,
-                }) => {
+export const Wizard = ({steps = defaultSteps, isShowBack= true, completed= {}, onDone = () => {}}) => {
+    const [activeStep, setActiveStep] = useState(0);
+    const [skipped, setSkipped] = useState(new Set());
 
-    const [state, setState] = useState({
-        stepWizard: {},
-    });
-    const [refresh, setRefresh] = useState(false);
+    const isStepOptional = (stepIndex) => {
+        return steps[stepIndex].optional;
+    };
 
-    const onStepChange = () => {
-        setRefresh(!refresh);
-        onNextStep();
+    const isStepSkipped = (stepIndex) => {
+        return skipped.has(stepIndex);
     };
 
-    const setInstance = stepWizard => {
-        setState({...state, stepWizard: stepWizard});
+    const isStepCompleted = (stepIndex) => {
+        return completed.has(stepIndex);
     }
 
-    return (
-        <Container className='container'>
-            <div className='jumbotron'>
-                <Row className={'justify-content-center'}>
-                    <Col>
-                        <StepWizard
-                            onStepChange={onStepChange}
-                            transitions={transitions}
-                            nav={showProgressBar && <WizardNav />}
-                            instance={setInstance}
-                            isHashEnabled={hashEnabled}
-                        >
-                            {children}
-                        </StepWizard>
-                    </Col>
-                </Row>
-            </div>
+    const handleNext = () => {
+        if (activeStep === steps.length - 1) {
+            onDone();
+        }
+        else {
+            let newSkipped = skipped;
+            if (isStepSkipped(activeStep)) {
+                newSkipped = new Set(newSkipped.values());
+                newSkipped.delete(activeStep);
+            }
+            setActiveStep((prevActiveStep) => prevActiveStep + 1);
+            setSkipped(newSkipped);
+        }
+    };
 
-            {<WizardController
-                stepWizard={state.stepWizard}
-                canProceed={canProceed}
-                skipButton={showSkipButton}
-                onComplete={onComplete}/>}
+    const handleBack = () => {
+        setActiveStep((prevActiveStep) => prevActiveStep - 1);
+    };
 
-        </Container>
-    );
-};
+    const handleSkip = () => {
+        if (!isStepOptional(activeStep)) {
+            throw new Error("You can't skip a step that isn't optional.");
+        }
+        if (activeStep === steps.length - 1) {
+            onDone();
+        }
+        else {
+            setActiveStep((prevActiveStep) => prevActiveStep + 1);
+            setSkipped((prevSkipped) => {
+                const newSkipped = new Set(prevSkipped.values());
+                newSkipped.add(activeStep);
+                return newSkipped;
+            });
+        }
```

Comment:
```suggestion
        if (activeStep === steps.length - 1) {
            onDone();
            return
        }
        setActiveStep((prevActiveStep) => prevActiveStep + 1);
        setSkipped(prevSkipped => new Set(prevSkipped).add(activeStep));
```

Reply:
Not a fan of empty return statements... 
Did the other part